### PR TITLE
Add WINDOWS_SUBSYSTEM constant

### PR DESCRIPTION
**Pull request by Kelimion**
true when -subsystem:windows for Windows targets, false otherwise.

### CODE REVIEW

**Comment by flysand7** [comment]
I think this should be an enum, not a bool.

image

There's more subsystems than just two, though what's true is that users will generally care about just two of them -- Console and Windows. With that I would like to argue that `Efi` is also a subsystem with viable use-case that can surface up when developing EFI bootloaders (OS development stuff).

Even though this value will be unchecked in anything other than console and windowed applications, I still think it's a good idea for consistency's sake..?

**Comment by Kelimion** [comment]
> There's more subsystems than just two, though what's true is that users will generally care about just two of them -- Console and Windows. With that I would like to argue that `Efi` is also a subsystem with viable use-case that can surface up when developing EFI bootloaders (OS development stuff).
> 
> Even though this value will be unchecked in anything other than console and windowed applications, I still think it's a good idea for consistency's sake..?

All of them are supported (by main.cpp's flag parser). Subsystem:Windows and not Subsystem:Windows. ;-)

File: src/checker.cpp [hunk 1]
```diff
@@ -973,6 +973,8 @@ gb_internal void init_universal(void) {
 	add_global_string_constant("ODIN_ROOT",    bc->ODIN_ROOT);
 	add_global_string_constant("ODIN_BUILD_PROJECT_NAME", bc->ODIN_BUILD_PROJECT_NAME);
 
+	add_global_bool_constant("WINDOWS_SUBSYSTEM", bc->use_subsystem_windows);
```
Comment (gingerBill) [inline src/checker.cpp:976]: It makes more sense to do this:
```
add_global_string_constant("ODIN_WINDOWS_SYBSYSTEM", the_string);
```

**Review by gingerBill — CHANGES_REQUESTED** [review]

File: src/checker.cpp (hunk 1 above)
Comment (gingerBill) [reply src/checker.cpp:976]: This also makes it work on non-Windows systems by having it be an empty string constant

Comment (Kelimion) [reply src/checker.cpp:976]: Sensible.

**Pull request merged**